lib/publiccloud/ssh_interactive.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Class with helpers related to SSH Interactive mode
+#
+# Maintainer: Pavel Dostal <pdostal@suse.cz>
+
+package publiccloud::ssh_interactive;
+use base opensusebasetest;
+use testapi;
+use Utils::Backends qw(set_sshserial_dev unset_sshserial_dev);
+use strict;
+use warnings;
+
+our @ISA    = qw(Exporter);
+our @EXPORT = qw(ssh_interactive_join ssh_interactive_part);
+
+sub ssh_interactive_join {
+    # Ensure the SSH tunnels is alive
+    assert_script_run "ps aux | grep -i '[s]sh -t -R'";
+
+    # Open SSH interactive session and check the serial console works
+    type_string("ssh " . get_var('PUBLIC_CLOUD_INSTANCE_IP') . "\n");
+    wait_serial("ssh_serial_ready", 10);
+
+    # Prepare the environment to use the SSH tunnel for upload/download from the worker
+    set_var('SUT_HOSTNAME',          get_var('PUBLIC_CLOUD_INSTANCE_IP'));
+    set_var('AUTOINST_URL_HOSTNAME', 'localhost');
+    set_sshserial_dev();
+
+    $testapi::distri->set_standard_prompt('root');
+}
+
+sub ssh_interactive_part {
+    # Leave the SSH interactive session
+    assert_script_run("if [[ -p /dev/$serialdev ]]; then true; else false; fi");
```

Comment:
Let's add to the comment, that we are checking here if we are still in the SSH session, it's obfuscated for someone who isn't that deep into it, that checking for pipe of `$serialdev` means we are in SSH.